lib/publiccloud/provider.pm

```diff
@@ -374,7 +374,7 @@ sub terraform_apply {
         $sle_version =~ s/-/_/g;
         my $ha_sap_repo = get_var('HA_SAP_REPO') ? get_var('HA_SAP_REPO') . '/SLE_' . $sle_version : '';
         my $suffix = sprintf("%04x", rand(0xffff));
-        my $fencing_mechanism = get_var('FENCING_MECHANISM', 'sbd');
+        my $fencing_mechanism = get_var('FENCING_MECHANISM', is_gce ? 'native' : 'sbd');
```

Comment:
Since the default for GCP and AWS is `native`:

- GCP: https://github.com/SUSE/ha-sap-terraform-deployments/blob/main/gcp/variables.tf#L412
- AWS: https://github.com/SUSE/ha-sap-terraform-deployments/blob/main/aws/variables.tf#L384

Meaning the outlier is Azure (https://github.com/SUSE/ha-sap-terraform-deployments/blob/main/azure/variables.tf#L464), why not do instead:

```
my $fencing_mechanism = get_var('FENCING_MECHANISM', is_azure ? 'sbd' : 'native');
```

And then you don't need to modify L14.